File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b) {
            
            return    a[0] < b[0];
            
        });

        int count = 1;
        for (int i = 1; i < points.size(); i++) {
            if (points[i][0] > points[i - 1][1]) {
                count++;
            } else {
                points[i][1] = min(points[i - 1][1], points[i][1]);
            }
        }

        return count;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: (x[1]))

        lastEnd = points[0][1]
        count = 1

        for i in range(1, len(points)):
            start, end = points[i]

            if start > lastEnd:
                count += 1
                lastEnd = end
            else:
                points[i][1] = lastEnd

        return count